pyHalo/realization_extensions.py

```diff
@@ -497,3 +502,212 @@ def _get_fluctuation_halos(realization, fluctuation_amplitude, fluctuation_size,
                              unique_tag=np.random.rand()) for i in range(len(amps))]
 
     return fluctuations
+
+def mask_annular(center_x, center_y, x_grid, y_grid, r_min, r_max = None):
+    """
+    An annular mask of specified inner and outer radii
+    
+    :param center_x: x-coordinate of center position of circular mask
+    :param center_y: y-coordinate of center position of circular mask
+    :param x_grid: x-coordinate grid
+    :param y_grid: y-coordinate grid
+    :param r_min: inner radius of mask in units of grid coordinates
+    :param r_max: outer radius of mask in units of grid coordinates
+    :return: mask array of shape x_grid with =0 inside the radius and =1 outside
+    """
+    x_shift = x_grid - center_x
+    y_shift = y_grid - center_y
+    R = np.sqrt(x_shift*x_shift + y_shift*y_shift)
+    mask = np.ones(x_shift.shape)
+    if r_max == None:
+        mask[R <= r_min] = 0
+    else:
+        mask[R <= r_min] = 0
+        mask[R >= r_max] = 0
+
+    return mask
+
+def convergence_map(realization, map_size, npix, lens_model_list_macro=None, kwargs_lens_macro=None, redshift_list_macro=None):
+    """
+    From a realization, this function computes the effective multiplane convergence. For further information, see the 
+    multiplane_convergence function documentation in utilities.py.
+
+    :return: the effective multiplane convergence, the two-dimensional x and y coordinate grids.
+    """
+    
+    kappa_map, _, _, _, _ = multiplane_convergence(realization, map_size/2, npix, lens_model_list_macro=None,
+                           kwargs_lens_macro=None, redshift_list_macro=None)
+    _R = np.linspace(-map_size/2, map_size/2, npix)
+    XX_, YY_ = np.meshgrid(_R, _R)
+
+    return kappa_map, XX_, YY_
+    
+def corr_kappa_with_mask(kappa_map, XX_, YY_, r, mu, apply_mask = True, r_min = 0, r_max = None, normalization = True): 
+    """
+    This function computes the two-point correlation function from a convergence map.
+    
+    :param kappa_map: the convergence map
+    :param XX_, YY_: the two-dimensional x and y coordinate grids
```

Comment:
same as above comment, it isn't clear from the documentation how XX_ and YY_ should relate to kappa_map. Can you instead pass in a map_size parameter, and then create the grids inside the function?